File: catbinary.py
def decode(message):
    message = int(message.replace("meow", "0").replace("MEOW", "1"), 2)
    return message.to_bytes((message.bit_length() + 7) // 8, "big").decode()


def encode(message):
    return (
        bin(int.from_bytes(message.encode(), "big"))
        .replace("b", "")
        .replace("0", "meow")
        .replace("1", "MEOW")
    )


i = input("Enter 'd' to decode or 'e' to encode a message to/from cat binary: ")
if i[0].lower() == "d":
    decodeMessage = input("Enter the message to decode: ")
    print("Decoded message: " + decode(decodeMessage))
elif i[0].lower() == "e":
    encodeMessage = input("Enter the message to convert to cat binary: ")
    print("Cat binary: " + decode(encodeMessage))

print("Thanks for using the cat binary translator!")
print("Remember to praise Schrodinger's Cat and have a great day.")

File: catbinary.js
function decode(message) {
    message = message.replace(/meow/g, '0').replace(/MEOW/g, '1')
    var result = "";
    var arr = message.match(/.{1,8}/g);
    for (var i = 0; i < arr.length; i++) {
        result += String.fromCharCode(parseInt(arr[i], 2).toString(10));
    }
    return result;
}

function encode(message) {
    var result = "";
    for (var i = 0; i < message.length; i++) {
        var bin = message[i].charCodeAt().toString(2);
        result += Array(8 - bin.length + 1).join("0") + bin;
    }
    return result.replace(/0/g, 'meow').replace(/1/g, 'MEOW');
}

let i = prompt("Enter 'd' to decode or 'e' to encode a message to/from cat binary: ")
if (i[0].toLowerCase() === "d") {
    let decodeMessage = prompt("Enter the message to decode: ")
    console.log("Decoded message: " + decode(decodeMessage))
}
else if (i[0].toLowerCase() === "e") {
    let encodeMessage = prompt("Enter the message to convert to cat binary: ")
    console.log("Cat binary: " + encode(encodeMessage))
}

console.log("Thanks for using the cat binary translator!")
console.log("Remember to praise Schrodinger's Cat and have a great day.")
